Firestore/core/src/firebase/firestore/util/hard_assert.cc

```diff
@@ -44,6 +47,33 @@ void Fail(const char* file,
 #endif
 }
 
+namespace {
+
+#if __APPLE__
```

Comment:
This (and also below) "pollute" this file and make it now somewhat platform dependent. (This has implications on the cmake tests too.) Some alternatives:
* separate failure_handler_callback into it's own .cc file.
* Let StdioFailureHandlerCallback be the unconditional default, and have iOS override that. But we'd need to ensure that happens prior to any firebase calls. The FIRFirestore ctor is the obvious place, but there's a couple factory methods that could be called prior to that too (and who knows what will be added in the future.) We could add the set call to the ctor of a global object... but ordering with other global objects that are also statically initialized isn't defined, so it would still be *possible* to end up calling the default.

Reply:
> +[FIRFirestore initialize]

*This* is what I was missing. Done.